beacon-chain/state/stategen/migrate.go

```diff
@@ -116,6 +116,7 @@ func (s *State) MigrateToCold(ctx context.Context, fRoot [32]byte) error {
 	}
 	if ok {
 		s.SaveFinalizedState(fSlot, fRoot, fInfo.state)
+		fInfo.state.SaveValidatorIndices() // used to handle Validator index invariant from EIP6110
```

Comment:
Is this the right spot?

Reply:
This is not the right spot, move it to block processing and refactor into helper line 172